File: interfaces/java/com/kx/Second.java
package com.kx;

import static com.kx.Type.NullInteger;
import static com.kx.Type.toTwoDigitString;

/**
 * This class is a kdb+ second.
 * 
 * @see <a href="https://code.kx.com/trac/wiki/Cookbook/InterfacingWithJava">Interfacing With Java at code.kx</a>
 * @author tech@kx.com
 */
public class Second {
	
	private int seconds;

	/**
	 * Constructs a kdb+ seconds value.
	 * 
	 * @param seconds
	 */
	public Second(int seconds) {
		this.seconds = seconds;
	}

	/**
	 * Converts to a string.
	 *
	 * @see java.lang.Object#toString()
	 */
	@Override
	public String toString() {
		if (seconds == NullInteger) 
				return "";
		else
				return new Minute(seconds / 60).toString() 
							+ ':' 
							+ toTwoDigitString(seconds % 60);
	}

	/**
	 * Checks for equality.
	 *
	 * @see java.lang.Object#equals(Object)
	 */
	@Override
	public boolean equals(Object obj) {
		if (obj instanceof Second) 
			return (seconds == ((Second)obj).seconds);
		else 
			return false;
	}

	/**
	 * Gets the hashcode.
	 *
	 * @see java.lang.Object#hashCode()
	 */
	@Override
	public int hashCode() {
		return seconds;
	}

	/**
	 * Gets the seconds.
	 * 
	 * @return the seconds
	 */
	public int getSeconds() {
		return seconds;
	}

	/**
	 * Sets the seconds.
	 * 
	 * @param seconds the new value
	 */
	public void setSeconds(int seconds) {
		this.seconds = seconds;
	}
	
}

File: interfaces/csharp/Kdbplus/Second.cs
namespace Kdbplus
{
    /// <summary>
    /// This is a kdb+ second value.
    /// </summary>
    public class Second
    {
        private int seconds;

        /// <summary>
        /// Constructs based on the given value.
        /// </summary>
        /// <param name="s">A value in seconds.</param>
        public Second(int s)
        {
            seconds = s;
        }

        /// <summary>
        /// Gets and sets the seconds value.
        /// </summary>
        public int TheSeconds
        {
            get { return seconds; }
            set { seconds = value; }
        }

        /// <summary>
        /// Converts to a string.
        /// </summary>
        /// <returns>The string representation of this value.</returns>
        public override string ToString()
        {
            return seconds == Type.NullInteger 
                ? "" 
                : new Minute(seconds/60).ToString() + ':' + Type.ToTwoDigitString(seconds%60);
        }
    }
}